modules/SofaBoundaryCondition/FixedConstraint.inl

```diff
@@ -245,27 +246,32 @@ void FixedConstraint<DataTypes>::projectJacobianMatrix(const core::MechanicalPar
 // When a new fixed point is added while its velocity vector is already null, projectVelocity is not usefull.
 // But when a new fixed point is added while its velocity vector is not null, it's necessary to fix it to null. If not, the fixed point is going to drift.
 template <class DataTypes>
-void FixedConstraint<DataTypes>::projectVelocity(const core::MechanicalParams* /*mparams*/, DataVecDeriv& /*vData*/)
+void FixedConstraint<DataTypes>::projectVelocity(const core::MechanicalParams* mparams, DataVecDeriv& vData)
 {
-#if 0 /// @todo ADD A FLAG FOR THIS
-    const SetIndexArray & indices = f_indices.getValue();
-    //serr<<"FixedConstraint<DataTypes>::projectVelocity, res.size()="<<res.size()<<sendl;
-    if( f_fixAll.getValue()==true )    // fix everyting
+    if(!f_activate_projectVelocity.getValue()) return;
+    const SetIndexArray & indices = this->f_indices.getValue();
+    helper::WriteAccessor<DataVecDeriv> res ( mparams, vData );
+
+    if( this->f_fixAll.getValue()==true )    // fix everyting
     {
         for( unsigned i=0; i<res.size(); i++ )
             res[i] = Deriv();
```

Comment:
ok to filter freeVelocity BUT
- it is not necessary to do it at each time step
- to be coherent it should also be done when all dofs are fixed
- the vector `freeVelocity` is not allocated in the general case
- if this projective constraint is applied on `freeVelocity`, every other projective constraints should also be applied.

The clean way is to call the `projectVelocity` visitor on `freeVelocity` where needed (in general in the solver that allocates `freeVelocity`) but definitively not hard-coded in `FixedConstraint::projectVelocity`.